### ISSUE TITLE
dynamic validation question

### ISSUE THREAD

**Issue by rileylnapier** [oreqizer/redux-forms#26]
hi, i have a situation in which i need dynamic validation.  specifically when country !== United States, then I need make another field required.

when I do this by adding/removing validation functions, the validate function doesn't get re-run.  

looking at componentWillReceiveProps, we only re-run validation when defaultValue change.

other places i see validate run, is onBlur.  

would you suggest me running the validation myself and calling fieldError action?  or do we like the idea of possibly running validation in componentWillReceiveProps?

**Comment by oreqizer** [comment]
I suggest you to keep the validation function the same, and the output based on `props`. What I mean is basically:

```js
class MyForm extends React.Component {
  handleValidate = value => {
    // whatever logic you need
    return this.props.country === 'UnitedStates' ? 'no US plz' : null;
  }

  render() {
    return (
      <Form name="form">
        <MyField name="field" validate={this.handleValidate} />
      </Form>
    );
  }
}
```

**Comment by rileylnapier** [comment]
can we have @field provide meta.formName?  this way in my InputField component I can do better validation without having to passaround formName into each component and into the Form?

**Comment by rileylnapier** [comment]
or better yet add props.actions?  or add field actions somehow to component wrapped with @field?

**Comment by rileylnapier** [comment]
i made this:  https://github.com/oreqizer/redux-forms/pull/27

**Comment by oreqizer** [comment]
Fixed by 1.0.0-7:
- `field` components now update on `validate` and `normalize` prop changes

Sorry for the delay! Let me know if this fixes your issue.

**Comment by rileylnapier** [comment]
looks good 👍
